flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogFunction.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Interface for a function in a catalog.
+ */
+public interface CatalogFunction {
+
+	/**
+	 * Get the source of the function. Right now, CLASS only.
+	 * @return the source of the function
+	 */
+	default Source getSource() {
```

Comment:
Any other source?
I don't quite understand the meaning of adding this. This method doesn't seem to work. Wait until we add other Sources?

Reply:
Yes, this is what we have in FLIP, but I think it's okay to leave this out.